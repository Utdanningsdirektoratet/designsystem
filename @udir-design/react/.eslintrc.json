{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
