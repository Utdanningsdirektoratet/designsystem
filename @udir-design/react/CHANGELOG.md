## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
