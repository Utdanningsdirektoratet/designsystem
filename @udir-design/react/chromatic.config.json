{
  "onlyChanged": true,
  "projectId": "Project:667e8f07bf467ff4403dfe77",
  "storybookBaseDir": "@udir-design/react",
  "zip": true
}
