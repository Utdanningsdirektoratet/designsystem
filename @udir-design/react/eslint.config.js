import { FlatCompat } from '@eslint/eslintrc';
// eslint-disable-next-line @nx/enforce-module-boundaries
import baseConfig from '../../eslint.config.js';
import storybook from 'eslint-plugin-storybook';
import eslintjs from '@eslint/js';
const { configs } = eslintjs;

const compat = new FlatCompat({
  baseDirectory: import.meta.dirname,
  recommendedConfig: configs.recommended,
});

export default [
  ...baseConfig,
  ...compat.extends('plugin:@nx/react'),
  ...storybook.configs['flat/recommended'],
  {
    ignores: ['!.storybook'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
    languageOptions: {
      parserOptions: {
        project: [
          '@udir-design/react/tsconfig.*?.json',
          '@udir-design/react/tsconfig.storybook.json',
        ],
      },
    },
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    // Storybook & docs-specific overrides
    files: ['**/*.stories.{ts,tsx}', '**/{.storybook,demo,docs}/**/*.{ts,tsx}'],
    rules: {
      '@nx/enforce-module-boundaries': [
        'error',
        {
          allowCircularSelfDependency: true,
        },
      ],
    },
  },
];
