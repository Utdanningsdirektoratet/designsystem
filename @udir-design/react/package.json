{
  "name": "@udir-design/react",
  "repository": {
    "type": "git",
    "url": "github:Utdanningsdirektoratet/designsystem",
    "directory": "@udir-design/react"
  },
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "version": "0.0.0-semantically-released",
  "type": "module",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    },
    "./alpha": {
      "import": "./dist/alpha.js",
      "require": "./dist/alpha.cjs"
    },
    "./beta": {
      "import": "./dist/beta.js",
      "require": "./dist/beta.cjs"
    },
    "./html": {
      "types": "./dist/html.d.ts"
    },
    "./style.css": "./dist/style.css"
  },
  "files": [
    "./dist"
  ],
  "scripts": {
    "build": "vite build",
    "inject": "pnpm i",
    "lint:cycles": "pnpm madge --circular ./src/alpha.ts",
    "test": "vitest --project unit",
    "typecheck": "tsc --build",
    "dev": "storybook dev --port 6005",
    "build:storybook": "storybook build --stats-json",
    "serve:storybook": "pnpm dlx http-server ./storybook-static",
    "test:storybook": "pnpm exec playwright install --with-deps chromium && pnpm vitest --project storybook"
  },
  "dependencies": {
    "@digdir/designsystemet-react": "1.5.0",
    "@navikt/aksel-icons": "^7.30.1",
    "@udir-design/css": "workspace:*",
    "tslib": "^2.8.1"
  },
  "devDependencies": {
    "@storybook/addon-a11y": "^9.1.5",
    "@storybook/addon-docs": "^9.1.5",
    "@storybook/addon-vitest": "^9.1.5",
    "@storybook/react-vite": "^9.1.5",
    "@types/diffable-html": "^5.0.2",
    "@types/mdast": "^4.0.4",
    "@types/ramda": "^0.30.2",
    "@types/react": "^19.1.13",
    "@types/react-dom": "^19.1.9",
    "@types/remark-heading-id": "^1.0.0",
    "@vitejs/plugin-react-swc": "^4.0.1",
    "@vitest/browser": "^3.2.4",
    "@vitest/coverage-v8": "^3.2.4",
    "@vitest/ui": "^3.2.4",
    "axe-core": "^4.10.3",
    "axe-playwright": "^2.2.2",
    "chromatic": "^13.1.4",
    "clsx": "2.1.1",
    "diffable-html": "^6.0.1",
    "eslint-plugin-storybook": "^9.1.5",
    "hast-util-from-html": "^2.0.3",
    "hast-util-is-element": "^3.0.0",
    "hast-util-to-html": "^9.0.5",
    "highcharts": "^12.4.0",
    "highcharts-react-official": "^3.2.2",
    "markdown-to-jsx": "^7.7.13",
    "mdast-util-to-hast": "^13.2.0",
    "playwright": "^1.55.0",
    "ramda": "^0.31.3",
    "react": "^19.1.1",
    "react-dom": "^19.1.1",
    "react-hook-form": "^7.62.0",
    "react-style-stringify": "^1.2.0",
    "remark-gfm": "^4.0.1",
    "remark-heading-id": "^1.0.1",
    "remark-parse": "^11.0.0",
    "remark-stringify": "^11.0.0",
    "sass": "^1.92.1",
    "storybook": "^9.1.5",
    "storybook-addon-pseudo-states": "^9.1.5",
    "storybook-addon-tag-badges": "^2.0.2",
    "unified": "^11.0.5",
    "unist-util-remove": "^4.0.0",
    "unist-util-visit": "^5.0.0",
    "vite": "^7.1.5",
    "vite-plugin-dts": "^4.5.4",
    "vite-tsconfig-paths": "^5.1.4",
    "vitest": "^3.2.4"
  },
  "peerDependencies": {
    "@types/react": "^18.3.7 || ^19.0.0",
    "react": "^18.3.1 || ^19.0.0",
    "react-dom": "^18.3.1 || ^19.0.0"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    }
  }
}
