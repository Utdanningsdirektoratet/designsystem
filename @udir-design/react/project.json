{
  "name": "@udir-design/react",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "@udir-design/react/src",
  "projectType": "library",
  "tags": [],
  "// targets": "to see all targets run: nx show project @udir-design/react --web",
  "targets": {
    "dev": {
      "continuous": true,
      "dependsOn": [
        "build",
        { "projects": "!css", "target": "^watch" },
        "lint:cycles"
      ]
    },
    "build:storybook": {
      "inputs": ["default", "^default", "{workspaceRoot}/README.md"],
      "dependsOn": ["build"],
      "cache": true,
      "outputs": ["{projectRoot}/storybook-static"]
    },
    "serve:storybook": {
      "continuous": true,
      "dependsOn": ["build:storybook"]
    },
    "test:storybook": {
      "cache": true,
      "dependsOn": ["build"]
    }
  }
}
