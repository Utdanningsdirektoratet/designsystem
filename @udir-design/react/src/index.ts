export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
