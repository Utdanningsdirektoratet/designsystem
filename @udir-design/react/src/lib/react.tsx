import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
