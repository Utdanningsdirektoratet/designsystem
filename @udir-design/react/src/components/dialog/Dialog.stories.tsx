import type { Meta, StoryObj } from '@storybook/react-vite';
import {
  Button,
  Heading,
  Paragraph,
  Textfield,
  Dialog,
  Field,
  Label,
  Suggestion,
  List,
  Textarea,
} from '@udir-design/react/alpha';
import { expect, userEvent, within } from 'storybook/test';
import { useRef, useState } from 'react';

const meta: Meta<typeof Dialog> = {
  component: Dialog,
  tags: ['alpha'],
  parameters: {
    customStyles: {
      display: 'grid',
      alignItems: 'start',
      justifyItems: 'center',
      story: {
        boxSizing: 'border-box',
        height: '100cqh',
        width: '100cqw',
      },
    },
    chromatic: {
      modes: {
        desktop: {
          viewport: { height: 1080 },
        },
      },
    },
  },
  play: async (ctx) => {
    // When not in Docs mode, automatically open the dialog
    const canvas = within(ctx.canvasElement);
    const button = canvas.getByRole('button');
    await userEvent.click(button);
    // Wait for dialog to fade in before running tests
    const dialog = canvas.getByRole('dialog');
    await new Promise<void>((resolve) => {
      dialog.addEventListener('animationend', () => {
        resolve();
      });
    });
  },
};

export default meta;
type Story = StoryObj<typeof Dialog>;

export const Preview: Story = {
  args: { closedby: 'any' },
  render: (args) => (
    <Dialog.TriggerContext>
      <Dialog.Trigger data-size={args['data-size']}>Åpne Dialog</Dialog.Trigger>
      <Dialog {...args}>
        <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
          Dialog header
        </Heading>
        <Paragraph style={{ marginBottom: 'var(--ds-size-2)' }}>
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis
          doloremque obcaecati assumenda odio ducimus sunt et.
        </Paragraph>
        <Paragraph data-size="sm">Dialog footer</Paragraph>
      </Dialog>
    </Dialog.TriggerContext>
  ),
  play: async (ctx) => {
    await meta.play?.(ctx);
    const canvas = within(ctx.canvasElement);
    const dialog = canvas.getByRole('dialog');
    const button = canvas.getByRole('button', {
      name: /åpne dialog/i,
    });

    await ctx.step('Dialog is open', async () => {
      await expect(dialog).toBeInTheDocument();
      await expect(dialog).toHaveAttribute('open');
    });

    await ctx.step('Dialog has content', async () => {
      const heading = within(dialog).getByRole('heading', {
        name: /dialog header/i,
      });
      const paragraph = within(dialog).getByText(
        /lorem ipsum dolor sit, amet consectetur adipisicing elit/i,
      );
      const footer = within(dialog).getByText(/dialog footer/i);
      await expect(heading).toBeInTheDocument();
      await expect(paragraph).toBeInTheDocument();
      await expect(footer).toBeInTheDocument();
    });

    await ctx.step('Close with blur', async () => {
      await userEvent.click(document.body);
      // Dialog has open="" when closed by blur
      await expect(dialog).toHaveAttribute('open', '');
    });

    await userEvent.click(button);
    await ctx.step('Close with close button', async () => {
      const closeButton = within(dialog).getByRole('button');
      await userEvent.click(closeButton);
      // Dialog does not have open attribute when closed by close button
      await expect(dialog).not.toHaveAttribute('open');
    });
    await userEvent.click(button);
  },
};

export const WithoutDialogTriggerContext: Story = {
  render(args) {
    const dialogRef = useRef<HTMLDialogElement>(null);
    return (
      <>
        <Button
          variant="secondary"
          onClick={() => dialogRef.current?.showModal()}
        >
          Åpne Dialog
        </Button>
        <Dialog {...args} ref={dialogRef}>
          <Paragraph data-size="sm">Dialog subtittel</Paragraph>
          <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
            Her bruker vi <code>ref</code> for å åpne dialogen
          </Heading>
          <Paragraph style={{ marginBottom: 'var(--ds-size-2)' }}>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis
            doloremque obcaecati assumenda odio ducimus sunt et.
          </Paragraph>
          Dialog footer
        </Dialog>
      </>
    );
  },
};

export const DialogWithOpenProp: Story = {
  render(args) {
    const [open, setOpen] = useState(false);

    return (
      <>
        <Button onClick={() => setOpen((prev) => !prev)}>
          Open Dialog with prop
        </Button>
        <Dialog {...args} open={open} onClose={() => setOpen(false)}>
          <Paragraph data-size="sm">Dialog subtittel</Paragraph>
          <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
            Dialog header
          </Heading>
          <Paragraph style={{ marginBottom: 'var(--ds-size-2)' }}>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis
            doloremque obcaecati assumenda odio ducimus sunt et.
          </Paragraph>
          Dialog footer
        </Dialog>
      </>
    );
  },
};

export const BackdropClosedbyAny: Story = {
  render() {
    const dialogRef = useRef<HTMLDialogElement>(null);

    return (
      <Dialog.TriggerContext>
        <Dialog.Trigger variant="secondary">Åpne Dialog</Dialog.Trigger>
        <Dialog ref={dialogRef} closedby="any">
          <Heading>
            Dialog med <code>closedby="any"</code>
          </Heading>
          <Paragraph>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis
            doloremque obcaecati assumenda odio ducimus sunt et.
          </Paragraph>
          <Paragraph data-size="sm">Dialog footer</Paragraph>
        </Dialog>
      </Dialog.TriggerContext>
    );
  },
};

export const WithHeaderAndFooter: Story = {
  render: () => (
    <Dialog.TriggerContext>
      <Dialog.Trigger>Gå til neste</Dialog.Trigger>
      <Dialog>
        <Dialog.Block>
          <Paragraph data-size="sm">Undertittel</Paragraph>
          <Heading>Dette må du vite før du går videre</Heading>
        </Dialog.Block>
        <Dialog.Block>
          <Paragraph style={{ marginBottom: 'var(--ds-size-2)' }}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
            sodales eros justo. Aenean non mi ipsum. Cras viverra elit nec
            vulputate mattis. Nunc placerat euismod pulvinar. Sed nec fringilla
            nulla, sit amet ultricies ante. Morbi egestas venenatis massa, eu
            interdum leo rutrum eu. Nulla varius, mi ac feugiat lacinia, magna
            eros ullamcorper arcu, vel tincidunt erat leo nec tortor. Sed ut dui
            arcu. Morbi commodo ipsum hendrerit est imperdiet imperdiet. Etiam
            sed maximus nisi. Quisque posuere posuere orci, non egestas risus
            facilisis a. Vivamus non tempus felis, in maximus lorem. Class
            aptent taciti sociosqu ad litora torquent per conubia nostra, per
            inceptos himenaeos.
          </Paragraph>
          <Paragraph>
            Etiam nec tincidunt est. Integer semper sodales efficitur.
            Pellentesque pellentesque varius leo id congue. Integer lacinia
            porttitor massa id euismod. Maecenas porta, magna nec interdum
            eleifend, risus magna condimentum neque, a gravida nisl risus a
            elit. Donec accumsan metus et lectus placerat varius. Donec
            tristique odio arcu. Donec cursus leo a dui auctor pulvinar. Sed in
            elit urna. Nunc vitae magna sed nibh elementum dignissim et ut
            massa.
          </Paragraph>
        </Dialog.Block>
        <Dialog.Block>Footer</Dialog.Block>
      </Dialog>
    </Dialog.TriggerContext>
  ),
};

export const DialogWithForm: Story = {
  render(args, context) {
    const dialogRef = useRef<HTMLDialogElement>(null);
    const [input, setInput] = useState('');

    return (
      <Dialog.TriggerContext>
        <Dialog.Trigger>Send inn skjema</Dialog.Trigger>
        <Dialog
          ref={dialogRef}
          onClose={() => setInput('')}
          closedby="any"
          {...args}
        >
          <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
            Skjemainnsending
          </Heading>
          <Textfield
            // @ts-expect-error We want the native "autofocus" and Reacts onMount smartness (see https://react.dev/reference/react-dom/components/input#input)
            autofocus="true"
            label="Navn"
            value={input}
            onChange={(e) => setInput(e.target.value)}
            id={context.id}
          />
          <div
            style={{
              display: 'flex',
              gap: 'var(--ds-size-2)',
              marginTop: 'var(--ds-size-4)',
            }}
          >
            <Button
              onClick={() => {
                window.alert(`Du har sendt inn skjema med navn: ${input}`);
                dialogRef.current?.close();
              }}
            >
              Send inn skjema
            </Button>
            <Button
              variant="secondary"
              onClick={() => dialogRef.current?.close()}
            >
              Avbryt
            </Button>
          </div>
        </Dialog>
      </Dialog.TriggerContext>
    );
  },
};

export const DialogWithMaxWidth: Story = {
  render: () => (
    <Dialog.TriggerContext>
      <Dialog.Trigger variant="secondary">Åpne Dialog</Dialog.Trigger>
      <Dialog style={{ maxWidth: 1200 }}>
        <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
          Dialog som er veldig bred
        </Heading>
        <Paragraph>
          Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis
          doloremque obcaecati assumenda odio ducimus sunt et.
        </Paragraph>
      </Dialog>
    </Dialog.TriggerContext>
  ),
};

const DATA_PLACES = [
  'Sogndal',
  'Oslo',
  'Brønnøysund',
  'Stavanger',
  'Trondheim',
  'Bergen',
  'Lillestrøm',
];

export const DialogWithSuggestion: Story = {
  render(ctx) {
    const dialogRef = useRef<HTMLDialogElement>(null);
    return (
      <Dialog.TriggerContext>
        <Dialog.Trigger variant="secondary">Åpne Dialog</Dialog.Trigger>
        <Dialog style={{ overflow: 'visible' }} ref={dialogRef}>
          <Dialog.Block>
            <Heading>Dialog med innhold utenfor</Heading>
          </Dialog.Block>
          <Dialog.Block>
            <Field>
              <Label>Velg en kommune</Label>
              <Suggestion>
                <Suggestion.Chips />
                <Suggestion.Input id={`${ctx.id}-input`} />
                <Suggestion.Clear />
                <Suggestion.List>
                  <Suggestion.Empty>Tomt</Suggestion.Empty>
                  {DATA_PLACES.map((place) => (
                    <Suggestion.Option key={place} label={place} value={place}>
                      {place}
                      <div>Kommune</div>
                    </Suggestion.Option>
                  ))}
                </Suggestion.List>
              </Suggestion>
            </Field>
          </Dialog.Block>
          <Dialog.Block
            style={{
              display: 'flex',
              gap: 'var(--ds-size-2)',
            }}
          >
            <Button
              onClick={() => {
                window.alert(`Skjema er sendt inn`);
                dialogRef.current?.close();
              }}
            >
              Send inn
            </Button>
            <Button
              variant="secondary"
              onClick={() => dialogRef.current?.close()}
            >
              Avbryt
            </Button>
          </Dialog.Block>
        </Dialog>
      </Dialog.TriggerContext>
    );
  },
  parameters: {
    a11y: {
      // TODO: these rules should be enabled after figuring out why they occur.
      // for some reason it says `aria-expanded` is not allowed
      config: {
        rules: [
          {
            id: 'aria-allowed-attr',
            enabled: false,
          },
          /* It does not like role="combobox" either */
          {
            id: 'aria-allowed-role',
            enabled: false,
          },
        ],
      },
    },
  },
};

export const DialogNonModal: Story = {
  parameters: {
    customStyles: {
      display: 'flex',
      height: '400px',
      alignItems: 'center',
      gap: 'var(--ds-size-4)',
    },
  },
  render(context) {
    const dialogRef = useRef<HTMLDialogElement>(null);
    return (
      <>
        <Field style={{ width: '400px' }}>
          <Label>Besvarelse</Label>
          <Textarea id={'textarea' + context.id} rows={8} />
        </Field>
        <Button variant="secondary" onClick={() => dialogRef.current?.show()}>
          Åpne skrivetips
        </Button>
        <Dialog
          ref={dialogRef}
          modal={false}
          style={{
            width: '300px',
            left: '80%',
          }}
        >
          <Heading style={{ marginBottom: 'var(--ds-size-2)' }}>
            Hva besvarelsen burde inneholde
          </Heading>
          <List.Unordered>
            <List.Item>Intro til prosjektet</List.Item>
            <List.Item>Oversikt over hva som er gjort i prosjektet</List.Item>
            <List.Item>Beskrivelse av prosjektmålene</List.Item>
            <List.Item>Metode for å oppnå prosjektmålene</List.Item>
            <List.Item>Leveranse</List.Item>
          </List.Unordered>
        </Dialog>
      </>
    );
  },
  play: async (ctx) => {
    await meta.play?.(ctx);
    const canvas = within(ctx.canvasElement);
    const dialog = canvas.getByRole('dialog');

    await ctx.step('Dialog is open', async () => {
      await expect(dialog).toBeInTheDocument();
      await expect(dialog).toHaveAttribute('open');
    });

    await ctx.step(
      'Interaction outside dialog when dialog is open',
      async () => {
        await userEvent.click(document.body);
        await expect(dialog).toHaveAttribute('open');
      },
    );
  },
};
