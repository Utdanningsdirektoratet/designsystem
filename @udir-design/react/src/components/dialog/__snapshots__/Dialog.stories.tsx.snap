// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Backdrop Closedby Any 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <h2>
    Dialog med
    <code>
      closedby="any"
    </code>
    og en veldig lang tittel
  </h2>
  <p data-variant="default">
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
  <p
    data-variant="default"
    data-size="sm"
  >
    Dialog footer
  </p>
</dialog>
`;

exports[`Dialog Non Modal 1`] = `
<button
  data-variant="primary"
  type="button"
>
  Open Dialog
</button>
<dialog
  data-modal="false"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <h2>
    Non-modal dialog
  </h2>
  <p data-variant="default">
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
</dialog>
`;

exports[`Dialog With Form 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <h2 style="margin-bottom: var(--ds-size-2);">
    Dialog med skjema
  </h2>
  <div>
    <label
      data-weight="medium"
      for="components-dialog--dialog-with-form"
    >
      Navn
    </label>
    <div>
      <input
        type="text"
        autofocus="true"
        id="components-dialog--dialog-with-form"
        value
      >
    </div>
  </div>
  <div style="display: flex; gap: var(--ds-size-4); margin-top: var(--ds-size-4);">
    <button
      data-variant="primary"
      type="button"
    >
      Send inn skjema
    </button>
    <button
      data-variant="secondary"
      type="button"
    >
      Avbryt
    </button>
  </div>
</dialog>
`;

exports[`Dialog With Max Width 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
  style="max-width: 1200px;"
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <h2 style="margin-bottom: var(--ds-size-2);">
    Dialog med en veldig lang bredde
  </h2>
  <p data-variant="default">
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
</dialog>
`;

exports[`Dialog With Open Prop 1`] = `
<button
  data-variant="primary"
  type="button"
>
  Open Dialog with prop
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <p
    data-variant="default"
    data-size="sm"
  >
    Dialog subtittel
  </p>
  <h2 style="margin-bottom: var(--ds-size-2);">
    Dialog header
  </h2>
  <p
    data-variant="default"
    style="margin-bottom: var(--ds-size-2);"
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
  Dialog footer
</dialog>
`;

exports[`Dialog With Suggestion 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
  style="overflow: visible;"
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <div>
    <h2>
      Dialog med suggestion
    </h2>
  </div>
  <div>
    <div>
      <label
        data-weight="medium"
        for="components-dialog--dialog-with-suggestion-input"
      >
        Velg en destinasjon
      </label>
      <div>
        <input
          type="text"
          list=":r0:"
          placeholder
          id="components-dialog--dialog-with-suggestion-input"
          value
          popovertarget=":r0:"
          aria-autocomplete="list"
          aria-controls=":r0:"
          aria-expanded="false"
          autocomplete="off"
          role="combobox"
        >
        <button
          data-icon="true"
          data-variant="tertiary"
          type="reset"
          aria-label="Tøm"
        >
        </button>
        <u-datalist
          data-sr-singular="%d forslag"
          data-sr-plural="%d forslag"
          id=":r0:"
          hidden
          role="listbox"
          tabindex="-1"
        >
          <u-option
            data-empty="true"
            role="none"
            tabindex="-1"
            aria-disabled="false"
            aria-selected="false"
          >
            Tomt
          </u-option>
          <u-option
            value="Sogndal"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Sogndal
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Oslo"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Oslo
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Brønnøysund"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Brønnøysund
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Stavanger"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Stavanger
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Trondheim"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Trondheim
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Bergen"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Bergen
            <div>
              Kommune
            </div>
          </u-option>
          <u-option
            value="Lillestrøm"
            tabindex="-1"
            role="option"
            aria-disabled="false"
            aria-selected="false"
          >
            Lillestrøm
            <div>
              Kommune
            </div>
          </u-option>
        </u-datalist>
      </div>
    </div>
  </div>
  <div>
    <button
      data-variant="secondary"
      type="button"
    >
      Avbryt
    </button>
  </div>
</dialog>
`;

exports[`Preview 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <h2 style="margin-bottom: var(--ds-size-2);">
    Dialog header
  </h2>
  <p
    data-variant="default"
    style="margin-bottom: var(--ds-size-2);"
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
  <p
    data-variant="default"
    data-size="sm"
  >
    Dialog footer
  </p>
</dialog>
`;

exports[`With Header And Footer 1`] = `
<button
  data-variant="primary"
  type="button"
  aria-haspopup="dialog"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <div>
    <p
      data-variant="default"
      data-size="sm"
    >
      Her er det også divider
    </p>
    <h2>
      Vi kan legge divider under header
    </h2>
  </div>
  <div>
    <p
      data-variant="default"
      style="margin-bottom: var(--ds-size-2);"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur sodales eros justo. Aenean non mi ipsum. Cras viverra elit nec vulputate mattis. Nunc placerat euismod pulvinar. Sed nec fringilla nulla, sit amet ultricies ante. Morbi egestas venenatis massa, eu interdum leo rutrum eu. Nulla varius, mi ac feugiat lacinia, magna eros ullamcorper arcu, vel tincidunt erat leo nec tortor. Sed ut dui arcu. Morbi commodo ipsum hendrerit est imperdiet imperdiet. Etiam sed maximus nisi. Quisque posuere posuere orci, non egestas risus facilisis a. Vivamus non tempus felis, in maximus lorem. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
    </p>
    <p data-variant="default">
      Etiam nec tincidunt est. Integer semper sodales efficitur. Pellentesque pellentesque varius leo id congue. Integer lacinia porttitor massa id euismod. Maecenas porta, magna nec interdum eleifend, risus magna condimentum neque, a gravida nisl risus a elit. Donec accumsan metus et lectus placerat varius. Donec tristique odio arcu. Donec cursus leo a dui auctor pulvinar. Sed in elit urna. Nunc vitae magna sed nibh elementum dignissim et ut massa.
    </p>
  </div>
  <div>
    Og over footer
  </div>
</dialog>
`;

exports[`Without Dialog Trigger Context 1`] = `
<button
  data-variant="primary"
  type="button"
>
  Open Dialog
</button>
<dialog
  data-modal="true"
  open
>
  <form method="dialog">
    <button
      data-icon="true"
      data-variant="tertiary"
      type="submit"
      aria-label="Lukk dialogvindu"
      data-color="neutral"
      name="close"
    >
    </button>
  </form>
  <p
    data-variant="default"
    data-size="sm"
  >
    Dialog subtittel
  </p>
  <h2 style="margin-bottom: var(--ds-size-2);">
    Dialog header
  </h2>
  <p
    data-variant="default"
    style="margin-bottom: var(--ds-size-2);"
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Blanditiis doloremque obcaecati assumenda odio ducimus sunt et.
  </p>
  Dialog footer
</dialog>
`;
